src/NumericalAlgorithms/Interpolation/Interpolate.hpp

```diff
@@ -0,0 +1,109 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <pup.h>
+
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "Domain/ElementId.hpp"
+#include "Evolution/EventsAndTriggers/Event.hpp"
+#include "Options/Options.hpp"
+#include "Parallel/CharmPupable.hpp"
+#include "Parallel/ConstGlobalCache.hpp"
+#include "Parallel/Invoke.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+class TimeId;
+namespace Tags {
+struct TimeId;
+template <size_t VolumeDim>
+struct Mesh;
+template <typename TagsList>
+struct Variables;
+}  // namespace Tags
+template <size_t Dim> class Mesh;
+template <size_t VolumeDim> class ElementIndex;
+namespace intrp {
+template <typename Metavariables>
+struct Interpolator;
+namespace Actions {
+struct InterpolatorReceiveVolumeData;
+}  // namespace Actions
+}  // namespace intrp
+/// \endcond
+
+namespace intrp {
+namespace Events {
+template <size_t VolumeDim, typename Tensors, typename EventRegistrars>
+class Interpolate;
+
+namespace Registrars {
+template <size_t VolumeDim, typename Tensors>
+struct Interpolate {
+  template <typename RegistrarList>
+  using f = Events::Interpolate<VolumeDim, Tensors, RegistrarList>;
+};
+}  // namespace Registrars
+
+/// Does an interpolation by calling Actions on Interpolator.
+template <size_t VolumeDim, typename Tensors,
+          typename EventRegistrars =
+              tmpl::list<Registrars::Interpolate<VolumeDim, Tensors>>>
+class Interpolate;  // IWYU pragma: keep
+
+template <size_t VolumeDim, typename... Tensors, typename EventRegistrars>
+class Interpolate<VolumeDim, tmpl::list<Tensors...>, EventRegistrars>
+    : public Event<EventRegistrars> {
+  /// \cond
+  explicit Interpolate(CkMigrateMessage* /*unused*/) noexcept {}
+  using PUP::able::register_constructor;
+  WRAPPED_PUPable_decl_template(Interpolate);  // NOLINT
+  /// \endcond
+
+  using options = tmpl::list<>;
+  static constexpr OptionString help =
+      "Starts interpolation by sending data to the Interpolator.";
+
+  Interpolate() = default;
+
+  using argument_tags =
+      tmpl::list<::Tags::TimeId, ::Tags::Mesh<VolumeDim>, Tensors...>;
+
+  template <typename Metavariables, typename ParallelComponent>
+  void operator()(const TimeId& time_id, const Mesh<VolumeDim>& mesh,
+                  const db::item_type<Tensors>&... tensors,
+                  Parallel::ConstGlobalCache<Metavariables>& cache,
+                  const ElementIndex<VolumeDim>& array_index,
+                  const ParallelComponent* const /*meta*/) const noexcept {
+    db::item_type<::Tags::Variables<tmpl::list<Tensors...>>> interp_vars(
```

Comment:
No reason to reference the tag here.  Just use `Variables<tmpl::list<Tensors...>>`.